ProvisionSample/Program.cs

```diff
@@ -588,29 +594,65 @@ static void Main(string[] args)
             return client;
         }
 
+        static async Task<IEnumerable<string>> GetTenantIdsAsync(string commonToken)
+        {
+            using (var httpClient = new HttpClient())
+            {
+                httpClient.DefaultRequestHeaders.Add("Authorization", "Bearer " + commonToken);
+                var response = await httpClient.GetStringAsync("https://management.azure.com/tenants?api-version=2016-01-29");
+                var tenantsJson = JsonConvert.DeserializeObject<JObject>(response);
+                var tenants = tenantsJson["value"] as JArray;
+
+                return tenants.Select(t => t["tenantId"].Value<string>());
+            }
+        }
+
         /// <summary>
         /// Gets an Azure access token that can be used to call into the Azure ARM apis. 
-        /// Requires configuring a Azure AD navtive applicatino which is then configured to delegated access rights to your Azure subscription resources
         /// </summary>
-        /// <returns></returns>
-        static string GetAzureAccessToken()
+        /// <returns>A user token to access Azure ARM</returns>
+        static async Task<string> GetAzureAccessTokenAsync()
         {
-            // Follow instructions here to setup your tenants provisioning app: https://azure.microsoft.com/en-us/documentation/articles/resource-group-create-service-principal-portal/#get-access-token-in-code
+            if (!string.IsNullOrWhiteSpace(azureToken))
+            {
+                return azureToken;
+            }
+
+            var commonToken = GetCommonAzureAccessToken();
+            var tenantId = (await GetTenantIdsAsync(commonToken.AccessToken)).FirstOrDefault();
```

Comment:
Is `FirstOrDefault()` used based on assumption that most people using the app only have a single tenant or is there always only 1 tenant by design of Azure?  I thought tenant was equivalent of directories and there could be multiple meaning we might select the tenant id of the one the consumer does not intend to use.